src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp

```diff
@@ -566,6 +567,26 @@ namespace OpenMS
         {
           dv = value;
         }
+
+        // Add unit *after* creating the term
+        if (unitAcc != "")
```

Comment:
this code actually used to drop the unit name since any assignment operator first clears the unit, so the code was broken before and should work now.